src/spikeinterface/core/node_pipeline.py

```diff
@@ -138,7 +138,100 @@ def compute(self, traces, start_frame, end_frame, segment_index, max_margin):
 
 # this is not implemented yet this will be done in separted PR
 class SpikeRetriever(PeakSource):
-    pass
+    """
+    This class is usefull to inject a sorting object in the node pipepline mechanisim.
+    It allows to compute some post processing with the same machinery used for sorting components.
+    This is a first step to totaly refactor:
```

Comment:
Docstring should be as static as possible...
```suggestion
    This is used by:
```